IBM/carbon-components-react
# fix(modal): make modal a focus trap

ref carbon-components [#1637](https://github.com/ibm/carbon-components/issues/1637), [#1075](https://github.com/ibm/carbon-components/issues/1075), [#1474](https://github.com/IBM/carbon-components/issues/1474)

Adds focus-trap package and implements a focus trap on the modal component. By default the focus trap is active, but users can disable if they _need to_. 

### changed
- implement `<FocusTrap>` component
- set initial focus to first interactable element (close button)

### to test

- observe that by default the modal is a focus trap. Tab moves from the close button to the first button and then to the second and loops back around. shift-tab cycles in reverse.

- in your dev environment set the `focusTrap` prop in Modal-story.js to false to see previous modal focus behavior of being able to move focus out of the modal.

- see if hitting escape while the modal has focus disables focus trapping on the modal

- see if clicking the "outer modal" or the area around the dialogue box does actually blur

attn: @carmacleod, @asudoh
